tests/func/test_dataset_query.py

```diff
@@ -459,6 +459,77 @@ def test_order_by_limit(cloud_test_catalog, save, animal_dataset):
     ]
 
 
+@pytest.mark.parametrize("save", [True, False])
+@pytest.mark.parametrize(
+    "cloud_type,version_aware",
+    [("s3", True)],
+    indirect=True,
+)
+def test_limit(cloud_test_catalog, save, animal_dataset):
+    catalog = cloud_test_catalog.catalog
+    q = DatasetQuery(animal_dataset.name, catalog=catalog).limit(2)
+    if save:
+        ds_name = "animals_cats"
+        q.save(ds_name)
+        result = DatasetQuery(name=ds_name, catalog=catalog).db_results()
+        dataset_record = catalog.get_dataset(ds_name)
+        assert dataset_record.status == DatasetStatus.COMPLETE
+    else:
+        result = q.db_results()
+
+    assert len(result) == 2
+    assert [posixpath.basename(r[3]) for r in result] == ["cat1", "cat2"]
+
+
+@pytest.mark.parametrize("save", [True, False])
+@pytest.mark.parametrize(
+    "cloud_type,version_aware",
+    [("s3", True)],
```

Comment:
Please don't use `s3` fixtures unless you are really testing the clouds-related functionality.